### ISSUE TITLE
Bounding box lost

### ISSUE THREAD

**Issue by HongyiDuanmu26** [nwojke/deep_sort#79]
Thank you for your great repo!
I have trained my own darkflow model and I am using your code to do tracking. I want to know why I found some bounding box lost in tracking while I can see those boxes when do detection with this yolo model.
It is not because of the confidence, the confidence is greater than 0.8. Is there any mechanism in algorithm to drop boxes?
Thank you very much!

**Comment by nwojke** [comment]
Maybe you can improve results by tuning the [cosine similarity threshold](https://github.com/nwojke/deep_sort/blob/master/deep_sort_app.py#L241), e.g., try a larger value.

**Comment by HongyiDuanmu26** [comment]
@nwojke Thank you for your reply! but I think this is not the answer I want. I have tried that and it turns out that this parameter just control the person re-identification part.
While technically, deep_sort has no business with detection part, I thought there will be no difference between the boxes in yolo and boxes in tracking. Also, I found that if I set FLAG.tracking false, which means just detection, this result is close to my yolo output.

**Comment by nwojke** [comment]
Conventionally, it is part of the trackers responsibility to filter false alarms and interpolate missed detections. In this case, the tracker will hold a track hypothesis for each detection in the sequence but it will only report those detections which have been confirmed (3 consecutive matched detections at first appearance). This is happening [here](https://github.com/nwojke/deep_sort/blob/master/deep_sort/track.py#L144). Feel free to play with the initialization logic if this is causing the error.

If it is not the cosine similarity threshold and not the initialization logic, then the Kalman filter configuration seems to not match your requirements. This would require you too increase the motion model uncertainty [here](https://github.com/nwojke/deep_sort/blob/master/deep_sort/kalman_filter.py#L52).

I hope this helps.

**Comment by HongyiDuanmu26** [comment]
Thank you so much for your reply! 
So my understanding is, while you don't use kalman filter for feature engineering in deep_sort, you still use kalman filter to verify the detection made by the detector. Am I right?
Thank you again!

**Comment by ParticularJ** [comment]
@nwojke Hello. Thank you for you great repo! I combine the Faster-Rcnn with deep_sort for detect and track the head in class. When the two people occlusion, the track id will exchanged. However, I watch your results  videos, it's pretty well. Can you give me some suggestion?  Forgive my bad English.

**Comment by nwojke** [comment]
@HongyiDuanmu26 You are correct. We still use the Kalman filter to validate associations based on motion.

**Comment by nwojke** [comment]
@ParticularJ Are you only tracking heads? The feature descriptor has been trained to discriminate people.

**Comment by ParticularJ** [comment]
@nwojke Thanks for your reply. Yes, I only tracking heads. But, all the heads is the one class, the algorithm faster-rcnn just detect head. Then, I want to use deep_sort tracking these heads.

**Comment by nwojke** [comment]
If you use head detections only then you likely need to re-train the feature descriptor.

**Comment by nwojke** [comment]
Sorry for the late reply. Tuning the Kalman filter is quite application specific and sadly nothing I can properly describe in an issue reply. If you are going down the road, it would be a great idea to read up on Kalman filters and how parameters influence the final result.

The visualization code contains everything that is necessary to visualize the posterior covariance of each track. See the commented out code [here ](https://github.com/nwojke/deep_sort/blob/master/application_util/visualization.py#L132) that we have used during parametrization.
